packages/fakta-fordel-beregningsgrunnlag/src/components/tilkommetAktivitet/TilkommetAktivitetAccordion.tsx

```diff
@@ -0,0 +1,149 @@
+import { Accordion, Label } from '@navikt/ds-react';
+import { ArbeidsgiverOpplysningerPerId, Beregningsgrunnlag } from '@navikt/ft-types';
+import { DDMMYYYY_DATE_FORMAT, ISO_DATE_FORMAT } from '@navikt/ft-utils';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import dayjs from 'dayjs';
+import { VurderInntektsforholdPeriode } from '@navikt/ft-types/src/beregningsgrunnlagFordelingTsType';
+import styles from './tilkommetAktivitetAccordion.less';
+import { erVurdertTidligere, slaaSammenPerioder } from './TilkommetAktivitetUtils';
+import VurdertIForrigeBehandlingIcon from '../felles/VurdertIForrigeBehandlingIcon';
+import TidligereVurderteAktiviteterPanel from './TidligereVurderteAktiviteterPanel';
+import TilkommetAktivitetField from './TilkommetAktivitetField';
+
+const formatDate = (date: string): string => (date ? dayjs(date, ISO_DATE_FORMAT).format(DDMMYYYY_DATE_FORMAT) : '-');
+
+const renderDateHeading = (fom: string, tom: string | undefined): ReactElement => {
+  if (!tom || tom === '9999-12-31') {
+    return (
+      <Label size="small">
+        <FormattedMessage id="BeregningInfoPanel.FordelBG.PeriodeFom" values={{ fom: formatDate(fom) }} />
+      </Label>
+    );
+  }
+  return (
+    <Label size="small">
+      <FormattedMessage
+        id="BeregningInfoPanel.FordelBG.PeriodeFomOgTom"
+        values={{
+          fom: formatDate(fom),
+          tom: formatDate(tom),
+        }}
+      />
+    </Label>
+  );
+};
+
+function finnFørsteFom(perioder: VurderInntektsforholdPeriode[]) {
+  return perioder.map(p => p.fom).sort((t1, t2) => (dayjs(t1).isBefore(dayjs(t2)) ? -1 : 1))[0];
+}
+
+function finnSisteTom(perioder: VurderInntektsforholdPeriode[]) {
+  return perioder.map(p => p.tom).sort((t1, t2) => (dayjs(t1).isAfter(dayjs(t2)) ? -1 : 1))[0];
+}
+
+function getHeading(perioder: VurderInntektsforholdPeriode[]) {
+  const fom = finnFørsteFom(perioder);
+  const tom = finnSisteTom(perioder);
+  return renderDateHeading(fom, tom);
+}
+
+interface TilkommetAktivitetAccordion {
+  beregningsgrunnlag: Beregningsgrunnlag;
+  arbeidsgiverOpplysningerPerId: ArbeidsgiverOpplysningerPerId;
+  formName: string;
+  index: number;
+  readOnly: boolean;
+  submittable: boolean;
+}
+
+const TilkommetAktivitetAccordion = ({
+  beregningsgrunnlag,
+  arbeidsgiverOpplysningerPerId,
+  formName,
+  index,
+  readOnly,
+  submittable,
+}: TilkommetAktivitetAccordion) => {
```

Comment:
Hmm, hvorfor interface og functions istedenfor en vanlig FunctionComponent med OwnProps egentlig?

Reply:
Ikkje noko besvisst valg. Tok kopi av eksisterende komponenter for tilkommet aktiviteter og der var det gjort slik. Tenkjer du at FunctionComponent er bedre?